### PR TITLE
Don't delete response collectors in a transaction

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

We no longer delete response collectors in a transaction after receiving a response to a "single" request, or receiving a "DONE" message in a stream. This fixes a possible error when loading 50+ answers in one query and then performing a second query.

## What are the changes implemented in this PR?

See https://github.com/vaticle/typedb-client-python/pull/250, which this PR is a copy of.

### CODE REVIEW

File: api/connection/TypeDBTransaction.ts [hunk 1]
```diff
@@ -40,9 +40,9 @@ export interface TypeDBTransaction {
```
Comment (alexjpwalker) [inline api/connection/TypeDBTransaction.ts:40]: Fixed a couple of method signatures that are incorrect as they are actually returning Promises

Comment (alexjpwalker) [inline api/connection/TypeDBTransaction.ts:40]: See comments on https://github.com/vaticle/typedb-client-python/pull/250/files, which this PR is a copy of.

Comment (alexjpwalker) [reply api/connection/TypeDBTransaction.ts:40]: But note, due to grpc-js printing out nonsense when calling `toString` on a gRPC message, I have NOT ported the improved "invalid request ID" message over to client-nodejs.

The output is just an array of numbers, they don't pretty-print the message.

File: stream/BidirectionalStream.ts [hunk 2]
```diff
@@ -63,11 +63,7 @@ export class BidirectionalStream {
         const responseQueue = this._responseCollector.queue(requestId);
         if (batch) this._dispatcher.dispatch(request);
         else this._dispatcher.dispatchNow(request);
-        try {
-            return await responseQueue.take() as Transaction.Res;
-        } finally {    
-            this._responseCollector.remove(requestId);
-        }
+        return await responseQueue.take();
```
Comment (flyingsilverfin) [inline stream/BidirectionalStream.ts:66]: i think this used to have `as Transaction.Res`?

**Review by flyingsilverfin — APPROVED** [review]

File: stream/BidirectionalStream.ts (hunk 2 above)
Comment (alexjpwalker) [reply stream/BidirectionalStream.ts:66]: Yeah I just noticed it was a redundant cast as the type of the thing being converted was already `Transaction.Res`.

**Pull request merged**